### DISCUSSION TITLE
Onnx connector and FunctionCalling

### DISCUSSION THREAD

**Discussion by MithrilMan** [microsoft/semantic-kernel#8013] [Q&A] [answered]
FunctionCalling is becoming a feature taken for granted.
Onnx connector are something very interesting if deeply supported, what's your plan about function calling for onnx connectors?
theoretically it should be something that depends on the model so I don't expect to be fully supported right now, but having a way to enable function calling on `OnnxRuntimeGenAIPromptExecutionSettings` could be a solution.

Do you plan integrating function calling like in OpenAI with ToolCallBehavior ?

**Comment by rogerbarreto** [comment, marked as answer]
Hi @MithrilMan, thanks for asking, this is a very desired feature across all of our connectors.
We plan on bringing them to those but don't have a defined ETA yet.
